x-pack/legacy/plugins/ml/public/jobs/new_job_new/recognize/resolvers.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import chrome from 'ui/chrome';
+import { i18n } from '@kbn/i18n';
+import { toastNotifications } from 'ui/notify';
+import { IPrivate } from 'ui/private';
+import { mlJobService } from '../../../services/job_service';
+import { ml } from '../../../services/ml_api_service';
+import { KibanaObjects } from './page';
+
+/**
+ * Checks whether the jobs in a data recognizer module have been created.
+ * Redirects to the Anomaly Explorer to view the jobs if they have been created,
+ * or the recognizer job wizard for the module if not.
+ */
+export function checkViewOrCreateJobs(
+  Private: IPrivate,
+  $route: any,
+  kbnBaseUrl: string,
+  kbnUrl: any
+) {
+  return new Promise((resolve, reject) => {
+    const moduleId = $route.current.params.id;
+    const indexPatternId = $route.current.params.index;
+
+    // Load the module, and check if the job(s) in the module have been created.
+    // If so, load the jobs in the Anomaly Explorer.
+    // Otherwise open the data recognizer wizard for the module.
+    // Always want to call reject() so as not to load original page.
+    ml.dataRecognizerModuleJobsExist({ moduleId })
+      .then((resp: any) => {
+        const basePath = `${chrome.getBasePath()}/app/`;
+
+        if (resp.jobsExist === true) {
+          const resultsPageUrl = mlJobService.createResultsUrlForJobs(resp.jobs, 'explorer');
+          window.location.href = `${basePath}${resultsPageUrl}`;
+          reject();
+        } else {
+          window.location.href = `${basePath}ml#/jobs/new_job/recognize?id=${moduleId}&index=${indexPatternId}`;
+          reject();
+        }
+      })
+      .catch((err: Error) => {
+        // eslint-disable-next-line no-console
+        console.log(`Error checking whether jobs in module ${moduleId} exists`, err);
```

Comment:
Should this be `console.error`? (I saw in another file `console.error` was used)